rosidl_default_generators/package.xml

```diff
@@ -9,32 +9,17 @@
   <author email="dthomas@osrfoundation.org">Dirk Thomas</author>
 
   <buildtool_depend>ament_cmake</buildtool_depend>
+  
+  <!-- Code generation for Actions depends on action_msgs -->
+  <buildtool_export_depend>action_msgs</buildtool_export_depend>
```

Comment:
Although this prevents downstream packages from having to declare a dependency on `action_msgs`, it now requires them to declare an export dependency on `rosidl_default_generators`. This is because packages building/running against a package that build an action may also need to link against (or import) `action_msgs`.

Maybe we could avoid having users export the `rosidl_default_generators` dependency if we avoid exposing `action_msgs` headers in public headers of generated interface libraries. I think there's only one place this is happening in the cpp generator. I'm not sure what the situation is in generated Python code though.

Reply:
#23 seems to resolve this issue by using rosidl_default_runtime to add an <exec_depend> on action_msgs.

Reply:
Yes, the check is happening [here](https://github.com/ros2/rosidl/blob/b8381d955a111cdf8a52a0f1891cc55de5f19db1/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake#L146-L151). IMO, it doesn't hurt to check. For instance, it's still possible someone will depend on `rosidl_core_generators` (or `rosidl_cmake`) directly. In both of those scenarios, the user still needs to depend on `action_msgs` explicitly if they have a custom action.

Reply:
> IMO, it doesn't hurt to check

Yes, now that I see the code that was used to check, it seems fine.

> If anything, we could consider updating the error message to suggest they depend on rosidl_default_generators instead shrug

That would be okay